packages/destination-actions/src/destinations/google-analytics-4/ga4-functions.ts

```diff
@@ -15,9 +15,9 @@ export function verifyParams(params: object | undefined): void {
   }
 
   Object.values(params).forEach((value) => {
-    if (typeof value != 'string' && typeof value != 'number') {
+    if (typeof value == 'object' || value instanceof Array) {
       throw new IntegrationError(
-        'GA4 only accepts string or number values for event parameters and item parameters. Please ensure you are not including null, array, or nested values.',
+        'GA4 does not accept null, array, or nested values for event parameters and item parameters. Please ensure you are using allowed data types.',
```

Comment:
Does `nested` equals to `object` in this case or should we include `object` in the error message as well?

Reply:
yeah we are referring to objects when we say nested maybe we should say object instead of nested? @kdaswani

Reply:
yup fine by me to change the wording to:

```suggestion
        'GA4 does not accept null, array, or object values for event parameters and item parameters. Please ensure you are using allowed data types.',
```